docs/source/nodes_and_pipelines/nodes.md

```diff
@@ -181,3 +181,72 @@ Out[2]: {'sum': 5}
 ```{note}
 You can also call a node as a regular Python function: `adder_node(dict(a=2, b=3))`. This will call `adder_node.run(dict(a=2, b=3))` behind the scenes.
 ```
+
+## How to use generator functions in a node
+
+Generator functions are useful for providing lazy iterators to search for items without storing in memory. We can wrap generator functions in a Kedro pipeline and save the result by calling the dataset's `save` method separately for each chunk.
+The following example will show you how to utilise `pandas chunksize` generator to process large datasets:
+
+We need to create a [custom dataset](https://kedro.readthedocs.io/en/stable/extend_kedro/custom_datasets.html) with `_load` having `chunksize` parameter for `pd.read_csv()` and `_save` should save the data in append-or-create mode, `a+`.
```

Comment:
Do we really need to change the `_load` method? Can't we just provide it in `load_args` in the catalog?

Could we show the code for the full custom dataset here? I suspect it just needs to be wrapped in a class like this:
```
class ChunkWiseCSVDataSet(pandas.CSVDataSet):
    ....
```

It would be nice if the user can follow this and test the functionality without too much guessing what needs to go where.

Reply:
Yes, it's a custom `pandas.CSVDataSet`.